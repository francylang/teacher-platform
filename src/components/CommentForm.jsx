import React, { Component } from 'react';

class CommentForm extends Component {
  render() {
    return (
      <article>
        <form action="" method="get">
          <label htmlFor="GET-comment">Comment:
            <input id="GET-comment" type="text" name="comment" />
          </label>
          <input className="submit-discussion" type="submit" name="submit" />
        </form>
      </article>
    );
  }
}

export default CommentForm;
