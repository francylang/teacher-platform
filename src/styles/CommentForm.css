.submit-comment {
  margin: 7.5px 0;
  padding: 3.5%;
  width: 100%;
  border: none;
  outline: none;
  background-color: #6c6ae5;
  box-shadow: 0 8px 12px 0 rgba(0, 0, 0, 0.13);
  border-radius: 5px;
  color: #fff;
  text-align: center;
  transition: background-color 0.5s;
  font-family: 'Ubuntu Mono';
  font-size: 115%; }
