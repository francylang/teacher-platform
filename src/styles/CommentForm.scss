.comment-form-containter {
  margin-top: 40px;
  display: grid;
  grid-template-columns: 1.5fr 2fr;
  grid-template-rows: 100%;
  grid-template-areas: "directions form";
}

.comment-form-directions {
  grid-area: directions;
  padding: 0 10px;
}

.comment-form-header {
  text-align: left;
  margin-top: 0;
}

.directions-text {
  text-align: left;
}

.comment-form {
  grid-area: form;
}

#get-comment {
  display: inline-block;
  margin-bottom: 7.5px;
  padding: 3.5%;
  width: 93%;
  box-shadow: 0 8px 12px 0 rgba(0,0,0,0.13);
  border-radius: 5px;
  border: none;
  color: #5D5B5E;
  outline: none;
  font-size: 120%;
  font-family: Nunito;
}

#get-comment::placeholder {
  color: #9A979D;
}

.submit-comment {
  padding: 2% 2.5%;
  width: 100%;
}
