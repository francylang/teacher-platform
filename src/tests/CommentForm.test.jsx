import React from 'react';
import ReactDOM from 'react-dom';
import CommentForm from '../components/CommentForm';
import { shallow } from 'enzyme';
import testSetup from './testSetup';

describe('<CommentForm />', () => {
  let wrapper;
  let mockFn = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<CommentForm />);
  });

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should render a comment form container', () => {
    const comment = wrapper.find('.comment-form-containter');

    expect(comment.length).toEqual(1);
  });

  it('should render an aside to hold comment form directions', () => {
    const aside = wrapper.find('.comment-form-directions');

    expect(aside.length).toEqual(1);
  });

  it('should render a header for the form', () => {
    const header = wrapper.find('.comment-form-header');

    expect(header.length).toEqual(1);
  });

  it('should render directions for the form', () => {
    const directions = wrapper.find('.directions-text');

    expect(directions.length).toEqual(1);
  });

  it('should render a form', () => {
    const form = wrapper.find('.comment-form');

    expect(form.length).toEqual(1);
  });

  it('should render an input', () => {
    const input = wrapper.find('#get-comment');

    expect(input.length).toEqual(1);
  });

  it('should render a submit button', () => {
    const submit = wrapper.find('.submit-comment');

    expect(submit.length).toEqual(1);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
})
