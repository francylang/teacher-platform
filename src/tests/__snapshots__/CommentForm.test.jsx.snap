// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CommentForm /> should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CommentForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <aside
          className="comment-form-directions"
>
          <h4
                    className="comment-form-header"
          >
                    Add a comment:
          </h4>
          <p
                    className="directions-text"
          >
                    Please be kind and specific, and provide links to any references resources. 
          </p>
</aside>,
        <form
          className="comment-form"
>
          <label
                    htmlFor="get-comment"
          >
                    <textarea
                              id="get-comment"
                              name="comment"
                              onChange={[Function]}
                              type="text"
                              value=""
                    />
          </label>
          <button
                    className="submit-comment"
                    name="submit"
                    onClick={[Function]}
                    type="submit"
          >
                    Submit
          </button>
</form>,
      ],
      "className": "comment-form-containter",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4
              className="comment-form-header"
>
              Add a comment:
</h4>,
            <p
              className="directions-text"
>
              Please be kind and specific, and provide links to any references resources. 
</p>,
          ],
          "className": "comment-form-directions",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Add a comment:",
              "className": "comment-form-header",
            },
            "ref": null,
            "rendered": "Add a comment:",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Please be kind and specific, and provide links to any references resources. ",
              "className": "directions-text",
            },
            "ref": null,
            "rendered": "Please be kind and specific, and provide links to any references resources. ",
            "type": "p",
          },
        ],
        "type": "aside",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              htmlFor="get-comment"
>
              <textarea
                            id="get-comment"
                            name="comment"
                            onChange={[Function]}
                            type="text"
                            value=""
              />
</label>,
            <button
              className="submit-comment"
              name="submit"
              onClick={[Function]}
              type="submit"
>
              Submit
</button>,
          ],
          "className": "comment-form",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <textarea
                id="get-comment"
                name="comment"
                onChange={[Function]}
                type="text"
                value=""
/>,
              "htmlFor": "get-comment",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "get-comment",
                "name": "comment",
                "onChange": [Function],
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "textarea",
            },
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit",
              "className": "submit-comment",
              "name": "submit",
              "onClick": [Function],
              "type": "submit",
            },
            "ref": null,
            "rendered": "Submit",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <aside
            className="comment-form-directions"
>
            <h4
                        className="comment-form-header"
            >
                        Add a comment:
            </h4>
            <p
                        className="directions-text"
            >
                        Please be kind and specific, and provide links to any references resources. 
            </p>
</aside>,
          <form
            className="comment-form"
>
            <label
                        htmlFor="get-comment"
            >
                        <textarea
                                    id="get-comment"
                                    name="comment"
                                    onChange={[Function]}
                                    type="text"
                                    value=""
                        />
            </label>
            <button
                        className="submit-comment"
                        name="submit"
                        onClick={[Function]}
                        type="submit"
            >
                        Submit
            </button>
</form>,
        ],
        "className": "comment-form-containter",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4
                className="comment-form-header"
>
                Add a comment:
</h4>,
              <p
                className="directions-text"
>
                Please be kind and specific, and provide links to any references resources. 
</p>,
            ],
            "className": "comment-form-directions",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Add a comment:",
                "className": "comment-form-header",
              },
              "ref": null,
              "rendered": "Add a comment:",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Please be kind and specific, and provide links to any references resources. ",
                "className": "directions-text",
              },
              "ref": null,
              "rendered": "Please be kind and specific, and provide links to any references resources. ",
              "type": "p",
            },
          ],
          "type": "aside",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                htmlFor="get-comment"
>
                <textarea
                                id="get-comment"
                                name="comment"
                                onChange={[Function]}
                                type="text"
                                value=""
                />
</label>,
              <button
                className="submit-comment"
                name="submit"
                onClick={[Function]}
                type="submit"
>
                Submit
</button>,
            ],
            "className": "comment-form",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <textarea
                  id="get-comment"
                  name="comment"
                  onChange={[Function]}
                  type="text"
                  value=""
/>,
                "htmlFor": "get-comment",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "id": "get-comment",
                  "name": "comment",
                  "onChange": [Function],
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "textarea",
              },
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Submit",
                "className": "submit-comment",
                "name": "submit",
                "onClick": [Function],
                "type": "submit",
              },
              "ref": null,
              "rendered": "Submit",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
